platform/api.progress/src/org/netbeans/api/progress/aggregate/BasicAggregateProgressFactory.java

```diff
@@ -53,6 +54,42 @@ public static ProgressContributor createProgressContributor(String trackingId) {
         return new ProgressContributor(trackingId);
     }
     
+    /**
+     * Create an aggregating progress ui handle for a long lasting task.
+     * @param contributors the initial set of progress indication contributors that are aggregated in the UI.
+     * @param allowToCancel either null, if the task cannot be cancelled or 
+     *          an instance of {@link org.openide.util.Cancellable} that will be called when user 
+     *          triggers cancel of the task.
+     * @param linkOutput an <code>Action</code> instance that links the running task in the progress bar
+     *                   to an output of the task. The action is assumed to open the apropriate component with the task's output.
+     * @param displayName to be shown in the progress UI
+     * @return an instance of <code>ProgressHandle</code>, initialized but not started.
+     * @since 1.59
+     */
+    public static AggregateProgressHandle createSystemHandle(String displayName, ProgressContributor[] contributors, 
```

Comment:
Probably a bit late to change, but: why the factory methods are in `ProgressHandle`, but not in `AggregateProgressHandle`?

Reply:
I could migrate them (add new ones, deprecated old ones), too - now when the API is being touched. Do you think it is beneficial ?